src/pygeometadata/mcf.py

```diff
@@ -0,0 +1,413 @@
+                f'received {type(keywords)} instead')
+
+        section_dict = {
+            'keywords': keywords,
+            'keywords_type': keywords_type
+        }
+
+        if vocabulary:
+            section_dict['vocabulary'] = vocabulary
+        self.mcf['identification']['keywords'][section] = section_dict
+
+    def describe_band(self, band_number, name=None, title=None, abstract=None,
+                      units=None):
+        """Define metadata for a raster band.
+
+        Args:
+            band_number (int): a raster band index, starting at 1
+            name (str): name for the raster band
+            title (str): title for the raster band
+            abstract (str): description of the raster band
+            units (str): unit of measurement for the band's pixel values
+        """
+        idx = band_number - 1
+        attribute = self.mcf['content_info']['attributes'][idx]
+        if name is not None:
+            attribute['name'] = name
+        if title is not None:
+            attribute['title'] = title
+        if abstract is not None:
+            attribute['abstract'] = abstract
+        if units is not None:
+            attribute['units'] = units
+
+        self.mcf['content_info']['attributes'].insert(idx, attribute)
+
+    def describe_field(self, name, title=None, abstract=None,
+                       units=None):
+        """Define metadata for a tabular field.
+        Args:
+            name (str): name and unique identifier of the field
+            title (str): title for the field
+            abstract (str): description of the field
+            units (str): unit of measurement for the field's values
+        """
+        def get_attr(attribute_list):
+            for idx, attr in enumerate(attribute_list):
+                if attr['name'] == name:
+                    return idx, attr
+            raise ValueError(
+                f'{self.datasource} has no attribute named {name}')
+
+        idx, attribute = get_attr(self.mcf['content_info']['attributes'])
+
+        if title is not None:
+            attribute['title'] = title
+        if abstract is not None:
+            attribute['abstract'] = abstract
+        if units is not None:
+            attribute['units'] = units
+
+        self.mcf['content_info']['attributes'].insert(idx, attribute)
+
+    def write(self):
+        """Write MCF to disk."""
+        with open(self.mcf_path, 'w') as file:
+            file.write(yaml.dump(self.mcf))
+        # TODO: always also write an ISO-191* XML doc?
+
+    def validate(self):
+        """Validate MCF against a jsonschema object."""
+        # validate against our own schema, which could
+        # be a superset of the core MCF schema.
+        # If we wanted to validate against core MCF,
+        # we could use pygeometa.core.validate_mcf
+        jsonschema.validate(self.mcf, MCF_SCHEMA)
+
+    def to_string(self):
+        pass
```

Comment:
Thanks for this tip! That was indeed the idea. It doesn't look like we detailed a use-case in the design doc though. I imagine it would be used for serialization, when that need arises.

An instance of an `MCF` includes a few attributes at the moment:
```python
        self.datasource = source_dataset_path
        self.mcf_path = f'{self.datasource}.yml'
        self.mcf = None  # the metadata dictionary
```

`a = MCF('data.tif')`

I could imagine that `str(a)`  would stringify all the attributes. But I could also imagine a use-case for just wanting to stringify `a.mcf`. So maybe `a.to_string()` would be an intuitive way to do that?

Anyway, I left this method empty because we don't have a use for it yet, so maybe I'll leave it like that for now?

Reply:
Sounds good to me!  Yes, settling on the use case for that would be very wise.  To further add to the list of possibilities for what `__str__` could do, sometimes classes call `__repr__` instead, which could be useful in debugging, something like `<MCF at 0x12345abcdef>`.  Anyways, yes, let's defer that until later.